app/presenters/tree_builder_storage.rb

```diff
@@ -24,8 +24,13 @@ def x_get_tree_roots(count_only, _options)
   end
 
   def x_get_tree_custom_kids(object, count_only, options)
-    return count_only ? 0 : [] if object[:id] != "global"
-    objects = MiqSearch.where(:db => options[:leaf]).visible_to_all
+    objects = MiqSearch.where(:db => options[:leaf])
+    objects = case object[:id]
```

Comment:
nice, then you can move this code to method in `app/models/miq_search.rb` to avoid duplicating code.

and you can do it in follow up PR for `tree_builder_vms_filter.rb`

```ruby
  objects = MiqSearch.where(:db => options[:leaf])
    objects = case object[:id]
              when "global" # Global filters
                objects.visible_to_all
              when "my"     # My filters
                objects.where(:search_type => "user", :search_key => User.current_user.userid)
   end
```

what do you think ?